project_generator_definitions/mcu/st/stm32f401xe.yaml

```diff
@@ -37,6 +37,8 @@ tool_specific:
             - null
             FlashDriverDll:
             - UL2CM3(-S0 -C0 -P0 -FD20000000 -FC1000 -FN1 -FF0STM32F4xx_512 -FS08000000 -FL080000 -FP0($$Device:STM32F401RETx$CMSIS\Flash\STM32F4xx_512.FLM))
+            TargetDriverDllRegistry_SetRegEntry_Name:
+            - -U0672FF484849785087111542 -O78 -S0 -C0 -A0 -N00("ARM CoreSight SW-DP") -D00(2BA01477) -L00(0) -TO18 -TC10000000 -TP21 -TDS8007 -TDT0 -TDC1F -TIEFFFFFFFF -TIP8 -FO7 -FD20000000 -FC1000 -FN1 -FF0STM32F4xx_512.FLM -FS08000000 -FL080000 -FP0($$Device:STM32F401RETx$CMSIS\Flash\STM32F4xx_512.FLM)
```

Comment:
Last question first: with my PR immidate flashing is possible: just export, unzip, open in keil, compile and flash - not necessary to open several config dialogs ... before flashing as before.
The shown minimal uvoptx file is enough for this new behaviour.

For me they differ too much that's why I didn't had a deeper look. But your idea was great. The existing FlashDriverDll option used in uvoptx works! I just tried it manually and used exactly the same option for a STM32F446RE. So maybe we don't need any new options except the debugger options nTsel, pMon and Key. I'll check it with more targets and debuggers. 

Unfortunately a nTsel is not enough i.e. without a SetRegEntry part the uvoptx file doesn't work.